src/lib/UXDController.ts

```diff
@@ -142,44 +156,50 @@ export class UXDController {
   }: {
     contractAddress: string;
     spender: string;
-    amount: BigNumber;
+    amount: number;
     signer: Signer;
   }): Promise<ContractTransaction> {
-    return ERC20__factory.connect(contractAddress, signer).connect(signer).approve(spender, amount);
+    const ethAmount = ethers.utils.parseEther(amount.toString());
+    return ERC20__factory.connect(contractAddress, signer).connect(signer).approve(spender, ethAmount);
   }
 
-  public allowance({
+  public async allowance({
     contractAddress,
     account,
     spender,
   }: {
     contractAddress: string;
     account: string;
     spender: string;
-  }): Promise<BigNumber> {
-    return ERC20__factory.connect(contractAddress, this.provider).allowance(account, spender);
+  }): Promise<number> {
+    const allowance = await ERC20__factory.connect(contractAddress, this.provider).allowance(account, spender);
+    return Number(ethers.utils.formatEther(allowance));
   }
 
-  public tokenBalance({
+  public async tokenBalance({
     contractAddress,
     account,
   }: {
     contractAddress: string;
     account: string;
-  }): Promise<BigNumber> {
-    return ERC20__factory.connect(contractAddress, this.provider).balanceOf(account);
+  }): Promise<number> {
+    const balance = await ERC20__factory.connect(contractAddress, this.provider).balanceOf(account);
+    return Number(ethers.utils.formatEther(balance));
   }
 
-  public uxdTotalSupply(): Promise<BigNumber> {
-    return this.uxdContract.totalSupply();
+  public async uxdTotalSupply(): Promise<number> {
+    const totalSupply = await this.uxdContract.totalSupply();
+    return Number(ethers.utils.formatEther(totalSupply));
   }
 
-  public mintedPerCollateral(token: string): Promise<BigNumber> {
-    return this.controllerContract.mintedPerCollateral(token);
+  public async mintedPerCollateral(token: string): Promise<number> {
```

Comment:
`getRedeemableAmountForCollateral` ?

Reply:
Oh it's the total amount ever minted?
Not just currently redeemable?
Is that useful?

Reply:
it could be. You don't want to know how much collateral has been deposited? For now it'll just be the amount of WETH (ETH + WETH) but we're tracking both asWETH.

Reply:
We haven't had a use for the total amount ever minted so far, I would remove it from the client at least

Reply:
Are there any other use-cases for this client **library** beyond the web frontend? If no, then this is just application code, not a library code, which reduces the scope, but let's be clear about what we're building. If this is an actual library, then there are other uses cases beyond the web frontend, and there might be some info that's not useful for the web frontend but useful to other library users (myself).

Reply:
If it's useful to you, then feel free to keep it of course!
I just don't see the use of this method (for everything we have built on Solana so far, we never needed this info) and want no useless code in the client